### DISCUSSION TITLE
Can't change RPGLE lint indent value

### DISCUSSION THREAD

**Discussion by Justin-E-Taylor** [codefori/vscode-ibmi#1394] [IBM i] [answered]
It seems no matter what I try, I keep getting this message on every line:
`Incorrect indentation. Expected 0, got {number here}`

I tried removing the "indent" property, as well as setting it to a crazy value.  It still wants a 0 indent.  Here's my current RPGLINT.JSON:
```
{
  "indent": 10,
  "PrototypeCheck": true,
  "NoOCCURS": true,
  "NoSELECTAll": true,
  "UppercaseConstants": true,
  "IncorrectVariableCase": true,
  "StringLiteralDupe": true,
  "NoSQLJoins": true,
  "PrettyComments": true,
  "NoGlobalSubroutines": true,
  "NoExternalTo": [
    "QCMD",
    "QP2TERM",
    "QSH",
    "SYSTEM",
    "QCMDEXC"
  ]
}
```

Any idea what I should check?

TIA

**Comment by Justin-E-Taylor** [comment]
Anyone?

**Comment by thomprl** [comment]
I haven't figured out exactly what changes it but press F1 and type indent and it should tell you to change the indent value.  Also there is a place in the bottom right that allows you to adjust the ident value.

**Comment by Justin-E-Taylor** [comment]
I think I'm setting it right, but it doesn't seem to have any effect.

In the Object Browser view, I right-click on my filter and select "Open RPG lint configuration".  The first value is for indent.  No matter what value I put in there, the linter assumes an indent of 0.  I've re-opened the source member and bounced VS Code.  It keeps giving the same warning.

```
{
  "indent": 2,
  "PrototypeCheck": true,
  "NoOCCURS": true,
  "NoSELECTAll": true,
  "UppercaseConstants": true,
  "IncorrectVariableCase": true,
  "NoSQLJoins": true,
  "PrettyComments": true,
  "NoGlobalSubroutines": true,
  "NoExternalTo": [
    "QCMD",
    "QP2TERM",
    "QSH",
    "SYSTEM",
    "QCMDEXC"
  ]
}
```

**Comment by thomprl** [comment]
Sorry Justin I’m not much help I don’t think I changed it in tjr linter. I was thinking that only affected the format document but not sure. But wanted to offer this up as well. Every other Friday there is a little group that gets together to discuss issues with code4i. Code for IBM i Fridays: A Free Seiden Community Eventseidengroup.comTodays meeting can be joined atJoin our Cloud HD Video Meetingus02web.zoom.usMeeting ID: 897 2338 2794Passcode: 596772On Jul 12, 2024, at 8:33 AM, Justin-E-Taylor ***@***.***> wrote:﻿
I think I'm setting it right, but it doesn't seem to have any effect.
In the Object Browser view, I right-click on my filter and select "Open RPG lint configuration".  The first value is for indent.  No matter what value I put in there, the linter assumes an indent of 0.  I've re-opened the source member and bounced VS Code.  It keeps giving the same warning.
{
  "indent": 2,
  "PrototypeCheck": true,
  "NoOCCURS": true,
  "NoSELECTAll": true,
  "UppercaseConstants": true,
  "IncorrectVariableCase": true,
  "NoSQLJoins": true,
  "PrettyComments": true,
  "NoGlobalSubroutines": true,
  "NoExternalTo": [
    "QCMD",
    "QP2TERM",
    "QSH",
    "SYSTEM",
    "QCMDEXC"
  ]
}

**Reply by worksofliam** [reply]
@thomprl @Justin-E-Taylor We are going to look at this in the meeting today :)

**Reply by thomprl** [reply]
Sorry my reply via email didn’t come through right. 

https://us02web.zoom.us/j/89723382794?pwd=Z0dtRUpSdm9rdVFVS3lOMXNFaStQdz09
Meeting ID: [897 2338 2794]
Passcode: 596772

**Reply by SJLennon** [reply]
@Justin-E-Taylor 
The indent value seems to be cached.  You need to change RPGLINT.JSON,  then disconnect, right click and connect using "Connect and Reload Server Setting".  That seems to have fixed it for me.

![image](https://github.com/user-attachments/assets/182d0046-35e5-4ef0-8807-436466162340)

**Reply by Justin-E-Taylor** [reply]
I tried the "Connect and Reload Server Settings" from my desktop, but it made no difference.  On my laptop, I installed Code for I this morning, so there shouldn't have been anything cached.

Just to clarify, it's the default indent that remains stuck at 0.  For nested blocks, the indent is correctly incremented by 2's from the default.

I tried a different library and created the RPGLINT.JSON.  The ident issue was there as well.

**Reply by worksofliam** [reply]
Can you confirm that your file is using Spaces and then Convert Tabs to Spaces?

<img width="1118" alt="Screenshot 2024-07-12 at 2 35 29 PM" src="https://github.com/user-attachments/assets/c18c9043-7373-4098-9188-c618e71ac740">

**Reply by SJLennon** [reply]
@Justin-E-Taylor 
I have a feeling the assumption is that code will always start in the first column, and that indentations are relative to that:
I imagine you are seeing something like this:
![image](https://github.com/user-attachments/assets/04d79a95-237d-4840-9334-b9103694063b)
When I Format Document the code is lined up at column 1:
![image](https://github.com/user-attachments/assets/366f8895-d7c9-48f2-968b-2d3eeac422d3)

**Reply by SJLennon** [reply]
@Justin-E-Taylor @worksofliam 
In the code above, I have 
![image](https://github.com/user-attachments/assets/008e706f-dc13-4b16-9c05-14bf5fa360ee)
The indent in RPGLINT.JSON, which is 2, overrides that.  I've always found it to work like this.

**Reply by Justin-E-Taylor** [reply]
So the indent value is for nested blocks, not for the origin, and the origin is always 1?

![image](https://github.com/user-attachments/assets/b58a93fa-993d-47b1-aeed-dc71cc8dfa1a)

**Reply by SJLennon** [reply, marked as answer]
Origin is always 1 in my experience (but I'm just a user).  Personally, I've not found a reason to start code in other than column 1.

**Reply by SJLennon** [reply]
And CRLF vs LF doesn't seem to make any difference.

**Reply by Justin-E-Taylor** [reply]
> Origin is always 1 in my experience (but I'm just a user). Personally, I've not found a reason to start code in other than column 1.

<facepalm>  You're correct.  For some reason, I assumed nested blocks were always indented 2 and the indent key sets the origin.

Code for i always has a gap between the line numbers and column 1.  In RDi, column 1 starts right up against the line numbers so I force a gap manually.


Thanks, everyone!  :)

**Comment by Justin-E-Taylor** [comment]
I won't be able to join that meeting.

I've deleted and re-created my RPGLINT.JSON.
I've tried different source members.
I've tried both my Windows desktop and Linux laptop.

I haven't tried a different library.  I'm going to try that next.

**Comment by thomprl** [comment]
I told you a little wrong as I'm just back at a PC.  Try this Press F1 and type Change Tab Display Size
